rust/connlib/snownet/src/node.rs

```diff
@@ -555,7 +555,11 @@ where
 
         if now >= next_reset {
             self.rate_limiter.reset_count();
-            self.next_rate_limiter_reset = Some(now + Duration::from_secs(1));
+            self.next_rate_limiter_reset = if self.connections.all_idle() {
+                None // If all connections are idle, there is no point in resetting the rate limiter.
```

Comment:
Hm, isn't there an edge case where we take too long to reset `rate_limiter` after all connections idle and a connection comes back alive?

Reply:
Hmm, I don't think so but now that I am looking at this more closely, I think this needs to change because we are using `get_or_insert` further above!